jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/admin/commands/TriggerNodeCommand.java

```diff
@@ -62,8 +65,16 @@ public Void execute(Context context) {
             if (node == null) {
                 throw new NodeNotFoundException("Node with id " + nodeId + " not found");
             }
-            NodeInstanceContainer nodeInstanceContainerNode =
-                    ((NodeInstanceContainer) wfp.getNodeInstance(wfp.getRuleFlowProcess().getParentNode(nodeId)));
+
+            Node parentNode = wfp.getRuleFlowProcess().getParentNode(nodeId);
+
+            // we need to unwrap the instance otherwise we are not getting the right nodeInstance
+            Map<String, Object> parentNodeMetaData = parentNode.getMetaData();
+            Boolean oldWrapValue = (Boolean) parentNodeMetaData.get(NodeInstance.METADATA_WRAP_ASYNC_NODE);
+            parentNodeMetaData.put(NodeInstance.METADATA_WRAP_ASYNC_NODE, false);
+            NodeInstanceContainer nodeInstanceContainerNode = (NodeInstanceContainer) wfp.getNodeInstance(parentNode.getId());
```

Comment:
There's a _NullPointerException_ in test _**ProcessInstanceAdminServiceImplTest#testTriggerSubProcessNodeInstance**_ due to this line returns null.

In that test, _parentNode.getId()_ corresponds to _node 2_:

_id: 2,  name= Hello_, being its container a _RuleFlowProcess_ (Ad-Hoc Sub-Process).

However, when retrieving the NodeInstanceContainer, as it's done in this line, with 

> wfp.getNodeInstance(parentNode.getId())

it's not found, because the node _"Hello"_ has:
_id=1, nodeId=2_

Therefore, it should match _nodeId_, instead of _id_.

Other possibility I've tested and worked out is using a filter like following suggestion replacing _getNodeInstance_ invocation:

```suggestion
           NodeInstanceContainer nodeInstanceContainerNode = (NodeInstanceContainer) wfp.getNodeInstances().stream().filter(ni -> ni.getNodeId() == parentNode.getId()).findFirst().orElse(null);
```